src/components/admin/event/EventDetailsForm/index.tsx

```diff
@@ -267,24 +277,48 @@ const EventDetailsForm = (props: IProps) => {
         {editing ? null : (
           <>
             <label htmlFor="cover">Cover Image</label>
-            <EventDetailsFormItem error={errors.cover?.message}>
+            <EventDetailsFormItem>
+              {coverUrl && (
+                <Image src={coverUrl} alt="Selected cover image" width={480} height={270} />
+              )}
               <input
                 type="file"
                 id="cover"
-                accept="image/png, image/gif, image/jpeg"
-                {...register('cover', {
-                  validate: cover => {
-                    const list = cover as FillInLater as FileList;
-                    if (list.length === 0) return 'Required';
-                    if (list.length > 1) return 'Only 1 image upload is allowed';
-                    if ((list.item(0)?.size ?? 0) >= config.file.MAX_EVENT_COVER_SIZE_KB * 1024) {
-                      return 'File is too large';
-                    }
-                    return true;
-                  },
-                })}
+                accept="image/*"
+                onChange={async e => {
+                  const file = e.currentTarget.files?.[0];
+                  e.currentTarget.value = '';
+                  if (file) {
+                    setSelectedCover(file);
+                  }
+                }}
               />
             </EventDetailsFormItem>
+            <Cropper
+              file={selectedCover}
+              aspectRatio={1920 / 1080}
+              maxFileHeight={1080}
+              maxSize={config.file.MAX_EVENT_COVER_SIZE_KB * 1024}
```

Comment:
I like this maxSize thing here as a warning for users!

one thing I think that's worth mentioning is that the max file size can vary between testing.api and prod (for testing it's 256 KB, for prod its 2048 KB), so on testing you might get some weird behavior:
- you try to upload an image that's in between 256 KB and 2048 KB
- the event is created successfully but the image doesn't get uploaded

this isn't a massive issue since it only affects testing and not prod, but it could be worth looking into having this config value switch depending on the deploy/runtime var

Reply:
just for reference:

on prod: 
- event cover: 2048
- pfp: 256
- banner (don't think we use these currently): 2048
- merchPhoto: 1024
- resume: 2048

on testing/staging:
- event cover: 256
- pfp: 256
- banner (don't think we use these currently): 256
- merchPhoto: 1024
- resume: wasn't specified in caprover for some reason

Reply:
This is handled by the config file which sets the max size depending on prod vs dev

Regardless, the cropper is guaranteed to close with an image file under the specified file size. If the image is too large, it'll try compressing it with lower and lower quality until it fits under the maximum

Reply:
oh hm not sure why I was having that bug earlier then. Do we want to prevent the event from being updated if the cover is too large?

Reply:
hmm what image were you using?

we could add a check in `AdminEventManager.createNewEvent` to prevent creating an event if the cover is too large, but `Cropper` should make the same check already

Reply:
Does `Cropper` block form submission if the cover is too large? if not, then we should probably enforce a check here before allowing submission

Reply:
> Does `Cropper` block form submission if the cover is too large? if not, then we should probably enforce a check here before allowing submission

yes, but I can add another check inside `createNewEvent` as a sanity check just in case